budgetportal/models.py

```diff
@@ -307,12 +307,19 @@ def get_govt_functions(self):
     def get_financial_year(self):
         return self.government.sphere.financial_year
 
-    def get_latest_department_instance(self):
+    def get_latest_department_instance(self, require_website_url=False):
         """ Try to find the department in the most recent year with the same slug.
         Continue traversing backwards in time until found, or until the original year has been reached. """
-        newer_departments = Department.objects.filter(government__slug=self.government.slug,
-                                                      government__sphere__slug=self.government.sphere.slug,
-                                                      slug=self.slug).order_by('-government__sphere__financial_year__slug')
+        if require_website_url:
```

Comment:
I think this should rather be two totally different methods:

`Department.get_latest_department_instance` and `Department.get_latest_department_url`

It doesn't really make sense to have a modifier argument to `Department.get_latest_department_instance` for an arbitrary field

Reply:
the code will then be simpler and more self-explanatory